pkg/perf/iperf/graphql_nodes.go

```diff
@@ -57,7 +57,9 @@ func (g *GraphQl) ListPublicNodes(ctx context.Context, nodesNum int, farmID, exc
 		excludeFarmCond = fmt.Sprintf(", farmID_not_eq: %d", excludeFarmID)
 	}
 
-	whereCond := fmt.Sprintf(`where: { publicConfig: {%s} %s %s }`, pubCond, farmCond, excludeFarmCond)
+	nodeUpReportInterval := time.Minute * 40
+	nodeUpInterval := time.Now().Unix() - 2*int64(nodeUpReportInterval.Seconds())
+	whereCond := fmt.Sprintf(`where: { updatedAt_gte: %d, AND: {power_isNull: true, OR: {power: {state_eq: Up, target_eq: Up}}}, publicConfig: {%s} %s %s }`, nodeUpInterval, pubCond, farmCond, excludeFarmCond)
```

Comment:
I just have a question. is updatedAt is actually updated on node uptime ?

Reply:
I also noticed that some nodes `power` object is completely not set in graphql ( == null) because there were never an event to set it up or down by a farmer bot.

A node that it's power state/traged were never updated has no data. The chain on the other hand is programmed to return the default when u do NodeGetPower if there is no value set.

Reply:
I checked the proxy status filter for it
- https://github.com/threefoldtech/tfgrid-sdk-go/blob/development/grid-proxy/pkg/nodestatus/nodestatus.go#L37